src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -726,6 +728,28 @@ def iot_hub_identity_remove(cmd, client, hub_name, identities, resource_group_na
     return lro.identity
 
 
+def iot_hub_identity_update(cmd, client, hub_name, identity_type, resource_group_name=None):
+    resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
+    hub = iot_hub_get(cmd, client, hub_name, resource_group_name)
+    hub_identity = None
+
+    if identity_type == IdentityUpdateType.none.value:
+        hub_identity = _build_identity(system=False, identities=None)
+    elif identity_type == IdentityUpdateType.system_assigned.value:
+        hub_identity = _build_identity(system=True, identities=None)
+    elif identity_type == IdentityUpdateType.user_assigned.value:
+        if not hub.identity.user_assigned_identities:
+            raise ArgumentUsageError('Hub {0} is not currently using any user-assigned identities.'.format(hub_name))
+        hub_identity = _build_identity(system=False, identities=hub.identity.user_assigned_identities)
+
```

Comment:
When both user_assigned_identities and system_assigned_identities are turned on, should they also support building in this situation?

Reply:
To clarify, this was based on the guideline here:
https://github.com/Azure/azure-cli/pull/17473
Specifically here: https://github.com/Azure/azure-cli/pull/17473/files#diff-8ba75e105324557e00d490758e775b654b82a478a1b936379504cc8adbd6223cR35

It mentions that this is essentially setting the `type` for the identity, so if the user passes `system_assigned` it will set the identity type to system assigned, and remove user-assigned identities if it has to.

For `user_assigned`, the opposite, keep the user-assigned and remove the system identity.

If the hub has user-assigned identities already, the command would just be `iot hub identity assign --system` then, right?

Reply:
I have left a comment to discuss this issue [comment link](https://github.com/Azure/azure-cli/pull/17473/files#r635738837) . Let's wait for the reply from @fengzhou-msft

Reply:
> If the hub has user-assigned identities already, the command would just be iot hub identity assign --system then, right?

@c-ryan-k Hi, as I discussed with @fengzhou-msft , if the rest of the service can support passing `system_assigned` and `user_assigned` at the same time , we can add support for this case
```
- Add system assigned identity only when user assigned identities exist
    ```
    az <resource> identity update ... --type "SystemAssigned, UserAssigned"
    ```
```
Could you please give it a try?

Reply:
@c-ryan-k Hi, I have discussed with @fengzhou-msft , and now we have made some changes to the design of managed identity:

1. Delete `identity update` command and modify `identity remove` to make it support removing all the user assigned identities.
For details: https://github.com/Azure/azure-cli/pull/17473#discussion_r635847630

2. When the value of `user_assigned` parameter is `[]`, `identity remove` will remove all the user assigned identities.
For details: https://github.com/Azure/azure-cli/pull/17473#discussion_r635854174

Reply:
Could you follow the new guideline to modify the code and fix the CI issue (it need to pull the code from the remote dev branch)?